src/api/DefaultAzureResourceProvider.ts

```diff
@@ -17,18 +17,27 @@ export class DefaultAzureResourceProvider implements AzureResourceProvider {
 
     getResources(subscription: AzureSubscription): Promise<AzureResource[] | undefined> {
         return callWithTelemetryAndErrorHandling(
-            'provideResources',
+            'defaultAzureResourceProvider.getResources',
             async (context: IActionContext) => {
                 const subContext = createSubscriptionContext(subscription);
                 const client = await createResourceClient([context, subContext]);
+
                 // Load more currently broken https://github.com/Azure/azure-sdk-for-js/issues/20380
+                const allResources = await uiUtils.listAllIterator(client.resources.list());
+
+                // dedupe resources to fix https://github.com/microsoft/vscode-azureresourcegroups/issues/526
+                const allResourcesDeduped: GenericResource[] = dedupeAzureResources(allResources);
+                context.telemetry.measurements.resourceCount = allResourcesDeduped.length;
 
-                const allResources: GenericResource[] = await uiUtils.listAllIterator(client.resources.list());
-                const appResources = allResources.map(resource => this.createAppResource(subscription, resource));
+                if (allResourcesDeduped.length !== allResources.length) {
+                    context.telemetry.properties.duplicateResources = 'true';
```

Comment:
Should we capture the # of duplicates too?

Reply:
I agree. I think it'll be a more compelling data point.